frontend/src/metabase/lib/expressions/recursive-parser.js

```diff
@@ -332,7 +333,11 @@ export const adjustBooleans = tree =>
         return [
           operator,
           ...operands.map((operand, index) => {
-            if (!Array.isArray(operand) || args[index] !== "boolean") {
+            if (
+              !Array.isArray(operand) ||
+              args[index] !== "boolean" ||
+              operator === "not"
```

Comment:
This exception is necessary so that `NOT X` (where `X` is a boolean field) isn't formatted as `NOT X = True` (redundant, and very confusing).

Reply:
I think we should wait for the backend PR on this one so we can remove this exception entirely. I added that block to rewrite calls like `case([boolean], 0, 1)` to `case([boolean] = True, 0, 1)` because of the same backend error:

![image](https://user-images.githubusercontent.com/653604/166784329-0d5758fb-ca28-443d-b0c8-a4a619814517.png)


![image](https://user-images.githubusercontent.com/653604/166782966-3ecff072-8407-47a7-b57a-553be144b9b1.png)

With your exception added, it makes the functionality inconsistent - `case(NOT [boolean], ...)` doesnt work but `case([boolean], ...)` does:


![image](https://user-images.githubusercontent.com/653604/166794246-aefdc27e-4031-4555-a334-a98caf3bb62d.png)


 We might as well remove it as it wasn't a particularly well thought out piece of code, especially since not all SQL servers support booleans (Oracle...). It should be behind a driver feature and/or have a normalizer on the backend